.circleci/config.yml

```diff
@@ -1,42 +1,217 @@
-      - checkout
+executors:
+  ubuntu_vm:
+    machine:
+      image: ubuntu-2204:2024.04.4
+      docker_layer_caching: true
 
-      # Download and cache dependencies
-      - restore_cache:
+commands:
+  start_services:
+    parameters:
+      services:
+        description: List of services to run
+        type: string
+    steps:
+      - run:
+          name: Copy config files
+          command: |
+            cp script/config/podman-compose.yml ./compose.yaml
+            cp -r script/config/examples/sentinel1 ./sentinel1
+            cp -r script/config/examples/sentinel2 ./sentinel2
+            cp -r script/config/examples/sentinel3 ./sentinel3
+            cp -r script/config/examples/twemproxy.yml ./twemproxy.yml
+            cp -r script/config/examples/tls/master.conf ./master.conf
+            cp -r script/config/examples/tls/replica1.conf ./replica1.conf
+            cp -r script/config/examples/tls/replica2.conf ./replica2.conf
+            cp -r .circleci/ca-circleci-cert.pem ./ca-circleci-cert.pem
+            cp -r .circleci/circleci.crt ./circleci.crt
+            cp -r .circleci/circleci.key ./circleci.key
+            cp -r script/config/examples/tls/sentinel1 ./sentinel1_tls
+            cp -r script/config/examples/tls/sentinel2 ./sentinel2_tls
+            cp -r script/config/examples/tls/sentinel3 ./sentinel3_tls
+      - run:
+          name: Start services
+          command: |
+            docker-compose up << parameters.services >>
+          background: true
+  install_ruby:
+    steps:
+      - restore_cache: # Download and cache dependencies
           keys:
-          - bundle-{{ .Environment.CACHE_VERSION }}-{{ checksum "Gemfile.lock" }}
-
+            - asdf-ruby-3.0.2-v1
+      - run:
+          name: Install Ruby 3.0.2
+          command: |
+            cd $HOME
+            if [ ! -d ".asdf" ]; then
+              git clone https://github.com/asdf-vm/asdf.git ~/.asdf --branch v0.14.0
+            fi
+            echo '. "$HOME/.asdf/asdf.sh"' > ~/.bashrc
+            source .bashrc
+            asdf plugin add ruby https://github.com/asdf-vm/asdf-ruby.git
+            asdf install ruby 3.0.2
+            echo 'ruby 3.0.2' > $HOME/project/.tool-versions
+      - save_cache:
+          paths:
+            - ../.asdf
+          key: asdf-ruby-3.0.2-v1
+      - *persist-to-workspace
+  install_dependencies:
+    steps:
+      - restore_cache: # Download and cache dependencies
+          keys:
+            - bundle-{{ .Environment.CACHE_VERSION }}-{{ checksum "Gemfile.lock" }}
       - run:
-          name: bundle install
+          name: Install dependencies
           command: |
             # Required for the license_finder
             # The Gemfile.lock includes nokogiri but bundler
             # would install native gem instead if not forced
-            bundle config set --local force_ruby_platform true
+            source $HOME/.bashrc
+            bundle config set --local force_ruby_platform 'true'
```

Comment:
we shouldn't need this but since it was there , I guess it can stay for now.

Reply:
Done at https://github.com/3scale/apisonator/pull/385/commits/b1054977693e2509b970d67ab9812ba101327327